pysmt/solvers/solver.py

```diff
@@ -333,6 +334,7 @@ def last_result(self):
         """
         return self._last_result
 
+    @clear_pending_pop
```

Comment:
Test are failing because you used the wrong order for decorators: the right way is:
```python
@property
@clear_pending_pop
def assertions(self):
```